.setup/bin/sample_courses/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""
+Contains all the global constant variables that are used 
+throughout the sample courses.
+None of the variables should be changed after they are set here.
+If you ever want to change the global variables in another file,
+you have to use import sample_courses and not from sample_courses import *,
+but its highly unrecommended.
+
+To prevent cicurlar imports, don't import any files in this package and if
+you need to import something then use import sample_courses.filename not from 
+sample_courses.filename import *
+"""
+
+# flake8: noqa
+from ruamel.yaml import YAML
+from submitty_utils import dateutils
+import argparse
+import json
+
+import os
+
+def parse_args():
+    """
+    Parses out the arguments that might be passed to this script as it's run as a commandline
+    application.
+    :return: parsed args from the argparse module
+    """
+    parser = argparse.ArgumentParser(
+        description="Sets up the sample courses as well as creating the necessary users for the "
+                    "course as needed. It reads in the courses.json and users.json files from the "
+                    ".setup/data directory to determine what courses/users are allowed and then "
+                    "either adds all or just a few depending on what gets passed to this script")
+
+    parser.add_argument("--db_only", action='store_true', default=False)
+    parser.add_argument("--no_submissions", action='store_true', default=False)
+    parser.add_argument("--no_grading", action='store_true', default=False)
+    parser.add_argument("--users_path", default=os.path.join(SETUP_DATA_PATH, "users"),
+                        help="Path to folder that contains .yml files to use for user creation. Defaults to "
+                             "../data/users")
+    parser.add_argument("--submission_url", type=str, default="",help="top level url for the website")
+    parser.add_argument("--courses_path", default=os.path.join(SETUP_DATA_PATH, "courses"),
+                        help="Path to the folder that contains .yml files to use for course creation. Defaults to "
+                             "../data/courses")
+    parser.add_argument("--install_dir", type=str, default="/usr/local/submitty", help="install path of submitty")
+    parser.add_argument("--data_dir", type=str, default="/var/local/submitty", help="data path of submitty")
+    parser.add_argument("course", nargs="*",
+                        help="course code to build. If no courses are passed in, then it'll use "
+                             "all courses in courses.json")
+    return parser.parse_args()
+
+# Start of global variables
+
+yaml = YAML(typ='safe')
+
+CURRENT_PATH = os.path.dirname(os.path.realpath(__file__))
+SETUP_DATA_PATH = os.path.join(CURRENT_PATH, "..", "..", "data")
```

Comment:
I don't know if this would do anything, but would it be more consistant with the rest of the file to declare this _after_ the args are parsed, and then use and `SETUP_DATA_PATH = os.path.join(SUBMITTY_REPOSITORY, '.setup/data')`

Reply:
The funny thing is we can't because the `parse_args` function relies on the SETUP_DATA_PATH variable for the default path.
Also, don't really think we need to really use `os.path.dirname(os.path.realpath(__file__))`, but its doing its job well for now because `SETUP_DATA_PATH = "/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/data"` is the absolute directory in the VM.
In this PR, my goal is to just split up this file into readable parts, then create another PR to then clean the code up, and lastly make another PR to make it more efficient